healthcheck.go

```diff
@@ -13,33 +17,94 @@ type healthService struct {
+	client                   *http.Client
 }
 
 func newHealthService(config *healthConfig) *healthService {
 	service := &healthService{config: config}
 	service.checks = []health.Check{
-		service.sampleCheck(),
+		service.writerCheck(),
+		service.contentResolverCheck(),
+		service.producerCheck(),
 	}
 	return service
 }
 
-func (service *healthService) sampleCheck() health.Check {
+func (service *healthService) buildHealthCheck() health.HealthCheck {
+	return health.HealthCheck{
+		SystemCode:  service.config.appSystemCode,
+		Name:        service.config.appName,
+		Description: service.config.appDesc,
+		Checks:      service.checks,
+	}
+}
+
+func (service *healthService) writerCheck() health.Check {
+	return health.Check{
+		BusinessImpact:   "Content relationships to packages will not be written / updated",
+		Name:             "Content collection Neo4j writer health check",
+		PanicGuide:       "https://dewey.ft.com/upp-content-collection-rw-neo4j.html",
+		Severity:         1,
+		TechnicalSummary: "Checks if the service responsible with writing content collections to Neo4j is healthy",
+		Checker: func() (string, error) {
+			return service.httpAvailabilityChecker(service.config.writerHealthURI)
+		},
+	}
+}
+
+func (service *healthService) contentResolverCheck() health.Check {
+	return health.Check{
+		BusinessImpact:   "No notifications will be created for the content in unfolded collections",
+		Name:             "Document store API health check",
+		PanicGuide:       "https://dewey.ft.com/document-store-api.html",
+		Severity:         1,
+		TechnicalSummary: "Checks if the service responsible with saving and retrieving content is healthy",
+		Checker: func() (string, error) {
+			return service.httpAvailabilityChecker(service.config.contentResolverHealthURI)
+		},
+	}
+}
+
+func (service *healthService) producerCheck() health.Check {
 	return health.Check{
-		BusinessImpact:   "Sample healthcheck has no impact",
-		Name:             "Sample healthcheck",
-		PanicGuide:       "https://dewey.ft.com/content-collection-unfolder.html",
+		BusinessImpact:   "No notifications will be created for the content in unfolded collections",
+		Name:             "Message producer health check",
+		PanicGuide:       "https://dewey.ft.com/document-store-api.html",
 		Severity:         1,
-		TechnicalSummary: "Sample healthcheck has no technical details",
-		Checker:          service.sampleChecker,
+		TechnicalSummary: "Checks if Kafka can be accessed through http proxy",
+		Checker: func() (string, error) {
+			return service.config.producer.ConnectivityCheck()
+		},
 	}
 }
 
-func (service *healthService) sampleChecker() (string, error) {
-	return "Sample is healthy", nil
+func (service *healthService) httpAvailabilityChecker(healthUri string) (string, error) {
+	req, err := http.NewRequest(http.MethodGet, healthUri, nil)
+	if err != nil {
+		msg := fmt.Sprintf("Error while creating http health check request: %v", err)
+		return msg, errors.New(msg)
```

Comment:
I just realised that I can't do that since I also need to return `msg`, not just a error containing the `msg`